### DISCUSSION TITLE
🐞 "failed to fetch oauth token" when get Python SDK using Dagger CLI

### DISCUSSION THREAD

**Discussion by jiedxu** [dagger/dagger#12183] [Help]
### What is the issue?

I'm trying to run `dagger develop --sdk=python` after the initiation with a name.

### Dagger version

dagger v0.13.6 (registry.dagger.io/engine:v0.13.6) darwin/arm64

### Steps to reproduce

```
dagger init --name=blah
```

```
dagger develop --sdk=python
```

### Log output

✔ connect 0.2s
✔ cache request: mkfile /schema.json 0.0s
✔ load cache: mkfile /schema.json 0.0s
✔ cache request: blob://sha256:9121d31121f4c30e725664dd17c07133b4cb49e375a465f39f740736d4e69c0b 0.0s
✔ load cache: blob://sha256:9121d31121f4c30e725664dd17c07133b4cb49e375a465f39f740736d4e69c0b 0.0s
✔ moduleSource(refString: "."): ModuleSource! 0.0s
✔ ModuleSource.kind: ModuleSourceKind! 0.0s
✔ ModuleSource.resolveContextPathFromCaller: String! 0.0s
✔ ModuleSource.resolveFromCaller: ModuleSource! 0.2s
✔ ModuleSource.asModule(engineVersion: "latest"): Module! 0.0s
✔ Module.sdk: String! 0.0s
✔ ModuleSource.resolveFromCaller: ModuleSource! 0.0s
✔ ModuleSource.sourceSubpath: String! 0.0s
✔ ModuleSource.withSDK(sdk: "python"): ModuleSource! 0.0s
✔ ModuleSource.withSourceSubpath(path: ".dagger"): ModuleSource! 0.0s
✔ ModuleSource.resolveFromCaller: ModuleSource! 0.1s
✘ ModuleSource.asModule(engineVersion: "latest"): Module! 0.7s
! failed to create module: select: failed to update codegen and runtime: failed to generate code: failed to call sdk module codegen: select: call function "Codegen": process "/runtime" did not complete successfully: exit code: 2

Error: failed to generate code: input: moduleSource.withContextDirectory.withSDK.withSourceSubpath.asModule resolve: failed to create module: select: failed to update codegen and runtime: failed to generate code: failed to call sdk module codegen: select: call function "Codegen": process "/runtime" did not complete successfully: exit code: 2

Stdout:
marshal: json: error calling MarshalJSON for type *dagger.GeneratedCode: input: container.from resolve: failed to resolve image ghcr.io/astral-sh/uv:0.4.26@sha256:7775c60dca9cc5827c36757c32c75985244d8f31447565fa8147e2b2e11ad280: failed to resolve source metadata for ghcr.io/astral-sh/uv:0.4.26@sha256:7775c60dca9cc5827c36757c32c75985244d8f31447565fa8147e2b2e11ad280: failed to authorize: failed to fetch oauth token: unexpected status from GET request to https://ghcr.io/token?scope=repository%3Aastral-sh%2Fuv%3Apull&service=ghcr.io: 403 Forbidden

**Comment by JerryBaboona** [comment]
Same issue with `dagger v0.14.0 (registry.dagger.io/engine:v0.14.0) darwin/arm64`

Trying to run the quickstart in the docs [here](https://docs.dagger.io/quickstart/daggerize) (w/o Dagger Cloud)

**Comment by fcbry** [comment]
> Same issue with `dagger v0.14.0 (registry.dagger.io/engine:v0.14.0) darwin/arm64`
> 
> Trying to run the quickstart in the docs [here](https://docs.dagger.io/quickstart/daggerize) (w/o Dagger Cloud)
(Go SDK)
~trying to authenticate to azurecr.io (Microsoft Azure Container Registry). 
Verified the credentials, if I do a `docker login` (locally) then it works, but it doesn't seem to be able to use the `docker login` credentials when run in a Github action.~

Works here, with azurecr.io both version 0.13.7 and 0.14.0. My mistake was: 

```
		container.WithRegistryAuth(registry, puser, registryPassword)
	}
	return m.PushDockerImage(ctx, container, identifier)
```
vs 
```
		container = container.WithRegistryAuth(registry, puser, registryPassword)
	}
	return m.PushDockerImage(ctx, container, identifier)
```
The error message isn't very clear as you might think, it's saying the credentials were wrong, as opposed to, "you didn't provide any credentials". Maybe you did something similar in your python code?
